### PR TITLE
docs: Update `2.5.0` changelog to reflect breaking change

### LINKED ISSUE #6835: Retroactively document breaking change with `default_environment` in `v2.5.0`
Per

- https://github.com/meltano/meltano/issues/6677#issuecomment-1266034410

We did not anticipate the `dev` default to be a breaking change, but since it is a breaking change, we should update in `CHANGELOG.md` and in release notes.

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes #6835

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -81,6 +81,12 @@ This project adheres to [Semantic Versioning](http://semver.org/) and [Keep a Ch
 
 ## v2.5.0 (2022-08-24)
 
+### BREAKING CHANGES
+
+- [#6582](https://github.com/meltano/meltano/issues/6582) Make `default_environment` a top-level setting
+
+  Running meltano environment add dev as a one-time fix will resolve the issue, even if no other changes are made and no environment-specific settings are added.
```
Comment (WillDaSilva) [inline CHANGELOG.md:88]:
```suggestion
  Running `meltano environment add dev` as a one-time fix will resolve the issue, even if no other changes are made and no environment-specific settings are added.
```

**Review by pandemicsyn — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -81,6 +81,12 @@ This project adheres to [Semantic Versioning](http://semver.org/) and [Keep a Ch
 
 ## v2.5.0 (2022-08-24)
 
+### BREAKING CHANGES
```
Comment (WillDaSilva) [inline CHANGELOG.md:84]: We should note that this was an unintentional breakage, that we don't intend for minor version bumps to cause any breaking changes, and that this section was added retrospectively.

**Review by WillDaSilva — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (tayloramurphy) [inline CHANGELOG.md:88]:
```suggestion
  Running meltano environment add dev as a one-time fix will resolve the issue, even if no other changes are made and no environment-specific settings are added.
  
  _Note: This breaking change was added inadvertently in this version. We do not intend to release breaking changes in minor versions and have added this notice retroactively. _
```

**Review by tayloramurphy — APPROVED** [review]

**Review by WillDaSilva — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -81,6 +81,14 @@ This project adheres to [Semantic Versioning](http://semver.org/) and [Keep a Ch
...
+
+- [#6582](https://github.com/meltano/meltano/issues/6582) Make `default_environment` a top-level setting
+
+  Running `meltano environment add dev` as a one-time fix will resolve the issue, even if no other changes are made and no environment-specific settings are added.
+
```
Comment (WillDaSilva) [inline CHANGELOG.md:89]:
```suggestion

```

**Pull request merged**